src/docs/arch-cursor.dox

```diff
@@ -417,4 +417,33 @@ Thus, in the presence of a memory tracker, uses
 of "stray" pointers will be detected.  Even without a memory tracker, uses of "stray" pointers into
 the freed storage will likely yield the overwritten bytes, and not the previously seen key or value.
 
+@section arch_random_cursor Random Cursor
+
+WiredTiger uses a pseudo-random algorithm when using random cursors with the goal to achieve a good
+random distribution and being performant. Only the \c cursor->next function can be used to fetch a random record.
+The current pseudo-random algorithm is as follows:
+
+1. Starting from the root page we descend randomly down using WiredTiger's random number generator (RNG) until we
+reach a leaf page. Once we found a leaf page, we go into different conditions to grab a record from the page. 
```

Comment:
Suggestion: "Once we find a leaf page, we then evaluate three different conditions that determine our next steps to grab a record from our page."

The conditions in this section can be listed as a indented bullet point list. Having a separate numbering system will also help readers differentiate between the three conditions and the rest of the algorithm. For example, something like:

- Condition one: ...
- Condition two: ...

Reply:
I have reorganised this portion let me know what you think.